plugins/aws/s3/src/hopeit/aws/s3/object_storage.py

```diff
@@ -359,9 +365,10 @@ async def list_objects(self, wildcard: str = "*") -> List[ItemLocator]:
         """
         wildcard = wildcard + SUFFIX
         n_part_comps = len(self.partition_dateformat.split("/"))
-
+        if self.partition_dateformat:
```

Comment:
Add a comment here or clarify in docstring, what is this intended to do?
Why is only recursive when there is a partition_dateformat configured?

Reply:
I have created https://github.com/hopeit-git/hopeit.aws/issues/6 to address this.